### ISSUE TITLE
Typo in search doesn't show results when using Algolia as the backend search

### ISSUE THREAD

**Issue by ericmulder** [WebDevStudios/wp-search-with-algolia#97]
**Describe the bug**
When using algolia as the backend search engine, I encountered the following bug. Although algolia gives the right (fuzzy) results back, they don’t show up. For example, if I search for ‘seearching a match’, algolia return the post_id (say id = 1) of ‘searching a match’, but then the WP_Query still executes something like `SELECT bla WHERE post_title LIKE '%seearching a match%' and post_id IN (1)` which return zero records.

This is because the `s` parameter still gets added to the final WP_Query.

A quick fix is to reset the `s` parameter, but this breaks other functions like `get_search_query()`

I reset the s parameter here:

```
function labelvier_algolia_should_filter_query($should_filter, $query) {
	if($should_filter) {
		$query->set('s', null);
	}
	return $should_filter;
}
add_filter('algolia_should_filter_query', 'labelvier_algolia_should_filter_query', 10, 2);
```

**To Reproduce**
1. Set algolia as the backend search engine
2. Search with a typo in the search query
3. You get no results

**Expected behavior**
I should get results based on the search query.

**Screenshots**

**Desktop (please complete the following information):**
 - Version 1.4.0
- Wordpress 5.5.1

**Comment by richaber** [comment]
Howdy @ericmulder, 

Specifically in regards to your description of the WP_Query that you cite above.

The `s` should indeed be set on WP_Query, that is how WP_Query knows a search is being performed.

The search SQL that is used in the WHERE clause of WP_Query is filtered in `[Algolia_Search::posts_search()](https://github.com/WebDevStudios/wp-search-with-algolia/blob/6332237e071c40bf622f2facb38a29908a504ea0/includes/class-algolia-search.php#L230-L232)`, which returns an empty string when `[Algolia_Search::should_filter_query()](https://github.com/WebDevStudios/wp-search-with-algolia/blob/6332237e071c40bf622f2facb38a29908a504ea0/includes/class-algolia-search.php#L77)` returns true.

That short circuits the part of the query that looks like `' AND (((wp_posts.post_title LIKE %seearching%)OR (wp_posts.post_excerpt LIKE %seearching%) OR (wp_posts.post_content LIKE %seearching%))) '` before the query is executed.

That said, picking up testing from where I left off in #98, using the same methodology and test content described there, I do not experience the issue described above.

I will search the default post and page content ("Hello World" and "Sample Page") for a word using an _obvious_ typo, that I know standard WordPress search will not find, and Algolia should find.

I deactivated WP Search with Algolia to test the standard WordPress search.

Searching for "eczample" with standard search yielded 0 results.

Searching for "hllo" with standard search yielded 0 results.

I reactivated WP Search with Algolia to test the typo tolerance.

Searching for "eczample" with WP Search with Algolia yielded 1 result (the "Sample Page").

Searching for "hllo" with WP Search with Algolia yielded 1 result (the "Hello World" post).

Those results match expectations.

To test pagination with this, I will search for a word that is a little "fuzzy", in this case, "beef". The WordPress "Sample Page" contains "been" which the Algolia API should surface in the WP Search with Algolia result set, but standard WordPress search should not. "beef" occurs in multiple posts due to my use of Any Ipsum to generate posts as described in #98, so that will give me pagination for testing.

I deactivated WP Search with Algolia to test the standard WordPress search.

Searching for "beef" with standard search yielded 20 results, and 2 pages.
On page 1 of the search results, Query Monitor's "Conditionals" tab indicates that `is_search()` is true.
On page 2 of the search results, Query Monitor's "Conditionals" tab indicates that `is_paged()` and `is_search()` are true.

I reactivated WP Search with Algolia to test the typo tolerance.
Searching for "beef" with WP Search with Algolia yielded 21 results, and 3 pages.
On page 1 of the search results, Query Monitor's "Conditionals" tab indicates that `is_search()` is true.
On page 2 of the search results, Query Monitor's "Conditionals" tab indicates that `is_paged()` and `is_search()` are true.
On page 3 of the search results, Query Monitor's "Conditionals" tab indicates that `is_paged()` and `is_search()` are true.
Page 3 revealed the default "Sample Page" was included in the search results, because the word "been" is similar to the word "beef" and typo tolerance surfaced that as a potential match.

The issue you describe appears to be specific to your installation.
I would suggest the same troubleshooting techniques described in #98, deactivating other plugins, and switching to a default Twenty* theme.
